src/portable/portable.c

```diff
@@ -695,17 +695,27 @@ static int install_chroot_dropin(
         if (!text)
                 return -ENOMEM;
 
-        if (endswith(m->name, ".service"))
+        if (endswith(m->name, ".service")) {
+                const char *os_release_source;
+
+                r = access("/etc/os-release", F_OK);
+                if (r < 0) {
+                        if (errno != ENOENT)
+                                return log_debug_errno(SYNTHETIC_ERRNO(errno), "Failed to write '%s': %m", dropin);
```

Comment:
Hmm, this error message seems misplaced. Also not SYNTHETIC_ERRNO.

Reply:
Yes I definitely lacked imagination with the log message :-) The SYNTHETIC_ERRNO is copied from somewhere else when handling errno, but might be the wrong thing to do - should it be passed directly?

Reply:
SYNTHETIC_ERRNO is a trick to pass a made-up errno value through `log_*_errno` so that it can be used with `return`. It is necessary because we store the passed errno value as `ERRNO=...` field in the journal and we have a rule that only errno value that were passed from somewhere else are supposed to be logged in this fashion. In other words, we store the errno in the log line in the caller but not in the log line in the callee. If we already have an error value, SYNTHETIC_ERRNO should not be used. It also doesn't make sense to use it without `return`.